pkg/scheduler/framework/plugins/README.md

```diff
@@ -0,0 +1,122 @@
+[`KubeSchedulerConfiguration`](https://kubernetes.io/docs/reference/scheduling/config/).
+This section will cover how you can add arguments to existing in-tree plugins.
+Let's assume the plugin is called `FooPlugin` and we want to add an optional
+integer parameter named `BarParam`.
+
+First, we need to define a struct type named `FooPluginArgs`:
+
+```go
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Defined in pkg/scheduler/apis/config/types_pluginargs.go.
+type FooPluginArgs struct {
+	// We embed a TypeMeta type within FooPluginArgs so as to let
+	// Kubernetes know to persist the object in the system.
+	metav1.TypeMeta
+	BarParam int32
+}
+```
+
+Similarly, define `FooPluginArgs` in `k8s.io/kube-scheduler/config/v1beta1/types_pluginargs.go`.
+This time, however, in order to allow implicit default values for arguments,
+the type of the struct's fields may be pointers; the zero value of a pointer
+field (nil) will make the framework to use the default value. Since `BarParam`
+is of type `int32` and since we want a non-zero default value:
+
+```go
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Defined in staging/src/k8s.io/kube-scheduler/config/v1beta1/types_pluginargs.go.
+type FooPluginArgs struct {
+	metav1.TypeMeta `json:",inline"`
+	BarParam *int32 `json:"barParam,omitempty"`
+}
+```
+
+For each `types_pluginargs.go` addition, remember to register the type in the
+corresponding `register.go`, which will allow the scheduler to recognize
+`KubeSchedulerConfiguration` values at parse-time.
+
+When a `KubeSchedulerConfiguration` object is parsed (happens in
+`cmd/kube-scheduler/app/options/options.go`), the scheduler will convert from
+the latter type to the former type, filling in the unspecified fields with
+defaults. Speaking of defaults, define `SetDefaults_FooPluginArgs` in
+`pkg/scheduler/apis/config/v1beta1/defaults.go` as follows:
+
+```go
+// v1beta1 refers to k8s.io/kube-scheduler/config/v1beta1.
+func SetDefaults_FooPluginArgs(obj *v1beta1.FooPluginArgs) {
+	if obj.BarParam == nil {
+		obj.BarParam = pointer.Int32Ptr(42)
+	}
+}
+```
+
+Remember to write tests in `defaults_test.go` (the test may not pass until you
+run code generation; more on that later).
+
+Next, we need to define validators to make sure the user's configuration and
+your default values are valid. To do this, add something like this in
+`pkg/scheduler/apis/config/validation/validation_pluginargs.go`:
+
+```go
+// From here on, FooPluginArgs refers to the type defined in pkg/scheduler
+// definition, not the kube-scheduler definition since we're dealing with
+// post-default values.
+func ValidateFooPluginArgs(args config.FooPluginArgs) *field.Error {
+	if args.BarParam < 0 && args.BarParam > 100 {
+		return field.Invalid(path.NewPath("barParam"), args.BarParam, "must be in the range [0, 100]")
+	}
+	return nil
+}
+```
+
+Remember to write tests in `validation_pluginargs_test.go` to test your
```

Comment:
I would move the instructions for code generation earlier (after defining the types) with a heading.